pymc3/distributions/multivariate.py

```diff
@@ -1940,6 +1943,75 @@ def _distr_parameters_for_repr(self):
         return ["mu"]
 
 
+class CARRV(RandomVariable):
+    name = "car"
+    ndim_supp = 1
+    ndims_params = [1, 2, 1, 1]
+    dtype = "floatX"
+    _print_name = ("CAR", "\\operatorname{CAR}")
+
+    def make_node(self, rng, size, dtype, mu, W, alpha, tau, sparse=False):
+        mu = at.as_tensor_variable(mu)
+
+        if sparse:
+            W_sparse = scipy.sparse.csr_matrix(W)
+            W = aesara.sparse.as_sparse_variable(W_sparse)
+        else:
+            W = at.as_tensor_variable(W)
+
+        tau = at.as_tensor_variable(tau)
+        if tau.ndim > 0:
+            tau = tau[:, None]
+
+        alpha = at.as_tensor_variable(alpha)
+        if alpha.ndim > 0:
+            alpha = alpha[:, None]
+
+        if tau.ndim == 0:
+            tau = tau * at.ones(mu.shape)
+        if alpha.ndim == 0:
+            alpha = alpha * at.ones(mu.shape)
```

Comment:
If `alpha` and `tau` are supposed to be scalars, this will need to be removed.

Regardless, it looks like these steps are trying to broadcast `tau` and `alpha`, but this kind of broadcasting is costly, because new arrays will be allocated.  If you want to broadcast, use something like `at.broadcast_to` (or `at.extra_ops.broadcast_to` for earlier versions of Aesara).

Reply:
@brandonwillard , When I wrote this code, I was guided by the original implementation of CAR distribution from PR #4504 where alpha and tau may be scalar or vector, optionally.  I was trying to keep this opportunity alive. As I understand it, this feature is only needed to reduce the amount of user code, so in principle we can make the parameters alpha and tau can only be a vector.
@ckrapu , could you comment this?

Reply:
To clarify: as I understand it, alpha and tau are vectors, but often all their elements are the same. So being able to specify a scalar instead of a vector may be useful.